python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_system.py

```diff
@@ -189,7 +189,7 @@ def __contains__(self, other):
                             "'BioSimSpace._SireWrappers.Atom', or 'BioSimSpace._SireWrappers.Residue'.")
 
         # Return whether the object is in the system.
-        return self._sire_object.contains(other._sire_object.molecule())
+        return other._sire_object.molecule().number() in self._mol_nums
```

Comment:
Are these `mol_nums` unique for each molecule, like `uuid`s? Otherwise, how does `BioSimSpace` keep track of renumbered molecules?

Reply:
Yes, they are unique for each molecule and we additionally enforce the constraint that the numbers are unique when combining molecules to form a new system, i.e. you can't add the same molecule to itself. (Calling `.copy()` on a molecule creates a copy with a new `MolNum`.)